src/DotNet/Emit/MethodBodyReaderBase.cs

```diff
@@ -218,9 +239,11 @@ protected MethodBodyReaderBase(DataReader reader)
 		/// </summary>
 		OpCode ReadOpCode() {
 			var op = reader.ReadByte();
-			if (op != 0xFE)
-				return OpCodes.OneByteOpCodes[op];
-			return OpCodes.TwoByteOpCodes[reader.ReadByte()];
+			if (op == 0xFE)
+				return OpCodes.TwoByteOpCodes[reader.ReadByte()];
+			if (op >= 0xF0 && op <= 0xFB)
+				return context?.GetExperimentalOpCode(op, reader.ReadByte()) ?? OpCodes.UNKNOWN2;
```

Comment:
What do you think about:

```csharp
		OpCode ReadOpCode() {
			var op = reader.ReadByte();
			if (op == 0xFE)
				return OpCodes.TwoByteOpCodes[reader.ReadByte()];
			if (op >= 0xF0 && op <= 0xFB && reader.BytesLeft >= 1) {
				if (context?.GetExperimentalOpCode(op, reader.ReadByte()) is OpCode opCode)
					return opCode;
				else
					reader.Position--;
			}
			return OpCodes.OneByteOpCodes[op];
		}
```

That would retain the old behavior in all cases except when there's a 2nd byte available in the stream and there's an actual custom opcode match. Either is fine with me (it doesn't matter for my use case) but I imagine this would be the least surprising from the average user's point of view.

Reply:
Yes that works too and will do almost the same thing (`GetExperimentalOpCode()` can return `null` even if the feature is enabled by the user in which case the code will now return a 1-byte opcode).